### ISSUE TITLE
docker registry: image deletion by name and tag

### ISSUE THREAD

**Issue by FStefanni** [docker/distribution-library-image#119]
Hi,

this is a feature request.

Currently it is possible to delete images by using the REST API of a private registry, only by specifying the digest (`<reference>`)

```
DELETE /v2/<name>/manifests/<reference>
```

This is inconvenient, since it will delete all the images with the same digest, whilst the expected behavior is to delete just one specific image (i.e. "un-tag" it) and to perform the actual deletion only if it is the last tag referring to that digest.

This is also the behavior implemented by the docker CLI interface when managing local images. So this is also a issue of uniformity.

So I propose to add another API, to implement the deletion via image name and tag. For example something like this:
```
DELETE /v2/<name>/manifests/<tag>
```

In case of an image without a tag (i.e. in cases where the `docker images` command displays "none" as a tag for an image), it should be possible to specify a "special none tag". For example something like this:
```
DELETE /v2/<name>/manifests/_
```

Will you consider this feature?

Regards

**Comment by milosgajdos** [comment]
Can you open an issue on https://github.com/distribution/distribution, please?
This is the repo for Docker **image** only.

**Comment by FStefanni** [comment]
Hi,

sure, thank you.

Regards.

**Comment by antoinetran** [comment]
Hi @milosgajdos is-it possible to reopen this ticket, since the issue in OCI Distribution has been fixed in v1.0.0 (https://github.com/distribution/distribution/pull/2169)? The latest release of docker image registry is v2.7.1 and it does not contain this fix. It would be great to have a v2.7.2 with this fix, and thus I suggest reopening this ticket! Thanks!

As of today, the only reliable way to remove a tag without removing the whole manifest, is for us, to backup the whole repository elsewhere with a whitelist, then remove the whole repository and restoring it. It's a long inefficient process!

**Comment by tianon** [comment]
@antoinetran this repository is what turns releases from https://github.com/distribution/distribution into image updates at https://hub.docker.com/_/registry, so what needs to happen first is a new release on https://github.com/distribution/distribution which includes https://github.com/distribution/distribution/pull/3427
